src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +249,169 @@ public void onTraversalStart() {
+                    0, sampleToReaderMap.size() - 1);
+
+            try {
+                importer = new GenomicsDBImporter(sampleToReaderMap, mergedHeader, intervals.get(0), importConfiguration);
+            } catch (IOException e) {
+                throw new UserException("Error initializing GenomicsDBImporter", e);
+            }
+
+            try {
+                importer.importBatch();
+                assert importer.isDone();
+            } catch (IOException e) {
+                throw new UserException("GenomicsDB importBatch raised exception", e);
+            }
+
+            closeReaders(sampleToReaderMap);
+
+        } else {
+            for (int i = 0; i < sampleNames.size(); i += batchSize) {
+
+                int upperBoundSampleIndex = min(sampleNames.size(), i+batchSize);
+                Map<String, FeatureReader<VariantContext>> sampleToReaderMap =
+                    getFeatureReaders(sampleNames, sampleNameToFileName, i, upperBoundSampleIndex);
+
+                logger.info("Importing batch " + i + " with " + batchSize + " samples");
+                long variantContextBufferSize = vcfBufferSizePerSample * batchSize;
+
+                GenomicsDBImportConfiguration.ImportConfiguration importConfiguration =
+                        createImportConfiguration(workspace, arrayName,
+                                variantContextBufferSize, segmentSize,
+                                "", "",
+                                i, upperBoundSampleIndex);
+
+                try {
+                    importer = new GenomicsDBImporter(sampleToReaderMap, mergedHeader, intervals.get(0), importConfiguration);
+                } catch (IOException e) {
+                    throw new UserException("Error initializing GenomicsDBImporter for batch " + i +
+                            " processing ", e);
+                }
+
+
+                try {
+                    importer.importBatch();
+                    assert importer.isDone();
+                } catch (IOException e) {
+                    throw new UserException("GenomicsDB importBatch raised exception", e);
+                }
+
+                closeReaders(sampleToReaderMap);
+            }
         }
     }
 
     @Override
     public Object onTraversalSuccess() {
-        logger.info("Import completed!");
-        return importer.isDone();
+        if (batchSize==DEFAULT_ZERO_BATCH_SIZE) {
+            logger.info("Import completed!");
+        } else {
+            logger.info("Import of all batches to GenomicsDB completed!");
+        }
+        return true;
     }
 
-    @Override
-    public void closeTool() {
-        for (Map.Entry<String, FeatureReader<VariantContext>> reader : sampleToVCFMap.entrySet()) {
+    /**
+     * Method to create feature readers for input files or GCS URLs
+     * in the current batch
+     *
+     * @param sampleNames  List of all sample names
+     * @param sampleNameToFileName  Sample name to file name mapping
```

Comment:
Document `lowerSampleIndex` and `upperSampleIndex` as well -- in particular, document whether they are inclusive or exclusive indices.